dev/service-registry/service-locator/src/main/scala/com/lightbend/lagom/gateway/NettyServiceGateway.scala

```diff
@@ -366,15 +367,23 @@ class NettyServiceGateway(lifecycle: ApplicationLifecycle, config: ServiceGatewa
   }
 
   private val bindFuture = server.bind(config.host, config.port).channelFutureToScala
-  lifecycle.addStopHook(() => {
-    for {
-      channel <- bindFuture
-      closed <- channel.close().channelFutureToScala
-    } yield {
-      eventLoop.shutdownGracefully(10, 10, TimeUnit.MILLISECONDS)
-      poolMap.asScala.foreach(_.getValue.close())
+
+  val unbindTimeout = coordinatedShutdown.timeout(CoordinatedShutdown.PhaseServiceUnbind)
+  coordinatedShutdown.addTask(
+    CoordinatedShutdown.PhaseServiceUnbind,
+    "unbind-netty-service-gateway"
+  ) {
+      () =>
+        {
+          (for {
+            channel <- bindFuture
+            closed <- channel.close().channelFutureToScala
+          } yield {
+            eventLoop.shutdownGracefully(100, unbindTimeout.toMillis - 100, TimeUnit.MILLISECONDS)
```

Comment:
I just noticed that this returns a `Future`, so there's no guarantee that this is actually complete when the task returns `Done`.